### ISSUE TITLE
how to set learning rate decay?

### ISSUE THREAD

**Issue by liuxiaotong15** [materialsvirtuallab/megnet#177]
hi, is it possible to set a learning rate decay in the model? if not, do you have better way to implement the decay besides loop {train 1 epoch once and save the model -> reset the lr} ... thank you.

**Comment by liuxiaotong15** [comment]
maybe diy a callback like ReduceLRUponNan？

**Comment by chc273** [comment]
Hi @liuxiaotong15 
You may try three things, the first is writing an optimizer that suits your need, and then recompile the model by “model.compile” and use your optimizer there. The second is what you proposed, by using a callback, although I don’t think you need to save the weights every epoch though. You can simply set the lr in the callback class. The last one is that you may try the LearningRateScheduler in tensorflow.keras.

**Comment by liuxiaotong15** [comment]
thanks
